Node/src/models/o365ConnectorCard.ts

```diff
@@ -0,0 +1,577 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+//
+// Microsoft Teams: https://dev.office.com/microsoft-teams
+//
+// Bot Builder Microsoft Teams SDK GitHub
+// https://github.com/OfficeDev/BotBuilder-MicrosoftTeams
+//
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License:
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// 'Software'), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED ''AS IS'', WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+
+import * as builder from 'botbuilder';
+import * as teams from './';
+import * as sprintf from 'sprintf-js';
+
+export class O365ConnectorCard implements builder.IIsAttachment {
+    protected data = {
+        contentType: 'application/vnd.microsoft.teams.card.o365connector',
+        content: <teams.IO365ConnectorCard> {}
+    };
+    
+    constructor(protected session?: builder.Session) {
+    }
+
+    public title(text: string|string[], ...args: any[]): this {
+        if (text) {
```

Comment:
I'm not sure consistency with BotBuilder is super important - it's possible, even likely, that Steve didn't think of Adrian's use case. I initially said that I said it was an odd use case, but Adrian just found an example where it wasn't, so I think it's a reasonable request to have tighter logic. 

I have created analogs of botbuilder's "create" methods that take large numbers of parameters with most of them optional -- when you call them you'd have to know to pass null instead of '' or you'd have a lot of JSON fields filled up with '' instead of not existing at all. Would that cause any bad side effects? E.g. do we render a field differently when its value exists and === '' versus not existing at all?